sdk/digitaltwins/digitaltwins/samples/Readme.md

```diff
@@ -0,0 +1,282 @@
+
+```Javascript Snippet:dt_component_lifecycle
+// Update Component1 by replacing the property ComponentProp1 value
+const digitalTwinId = "myTwin";
+const componentPath = "Component1";
+const jsonPatch = {
+  ComponentProp1: "value2"
+};
+await serviceClient.updateComponent(digitalTwinId, componentPath, jsonPatch);
+```
+
+### Get digital twin components
+
+Get a component by providing name of a component and Id of digital twin to which it belongs.
+
+```Javascript Snippet:DigitalTwinsSampleGetComponent
+const digitalTwinId = "myTwin";
+const componentPath = "Component1";
+const component = await serviceClient.getComponent(digitalTwinId, componentPath);
+console.log(`Updated component: ${component}`);
+```
+
+## Create and list digital twin relationships
+
+### Create digital twin relationships
+
+`upsertRelationship` creates a relationship on a digital twin provided with Id of a digital twin, name of relationship such as "contains", Id of an relationship such as "FloorContainsRoom" and an application/json relationship to be created. Must contain property with key "\$targetId" to specify the target of the relationship. Sample payloads for relationships can be found [here](https://github.com/Azure/azure-sdk-for-net-pr/blob/feature/IoT-ADT/sdk/iot/Azure.Iot.DigitalTwins/samples/DigitalTwinServiceClientSample/DTDL/Relationships/HospitalRelationships.json "RelationshipExamples").
+
+```Javascript Snippet:dt_scenario
+const hospitalRelationships = [
+  {
+    $relationshipId: "BuildingHasFloor",
+    $sourceId: "BuildingTwin",
+    $relationshipName: "has",
+    $targetId: "FloorTwin",
+    isAccessRestricted: false
+  }];
+for (const relationship of hospitalRelationships) {
+  await serviceClient.upsertRelationship(
+    relationship["$sourceId"],
+    relationship["$relationshipId"],
+    relationship
+  );
+}
+```
+
+### List digital twin relationships
+
+`listRelationships` and `listIncomingRelationships` lists all the relationships and all incoming relationships respectively of a digital twin.
+
+```Javascript Snippet:dt_relationships_list
+const digitalTwinId = "myTwin";
+const relationships = serviceClient.listRelationships(digitalTwinId);
+for await (const relationship of relationships) {
+  console.log(`EventRoute: ${relationship}`);
+}
+```
+
+```Javascript Snippet:dt_incoming_relationships_list
+const digitalTwinId = "myTwin";
+const incomingRelationships = serviceClient.listIncomingRelationships(digitalTwinId);
+for await (const incomingRelationship of incomingRelationships) {
+  console.log(`EventRoute: ${incomingRelationship}`);
+}
+```
+
+## Create, list, and delete event routes of digital twins
+
+### Create event routes
+
+To create an event route, provide an Id of an event route such as "sampleEventRoute" and event route data containing the endpoint and optional filter like the example shown below.
```

Comment:
update to "myEventRouteId" per your sample below?

Reply:
Fixed

---
In reply to: [455393986](https://github.com/Azure/azure-sdk-for-js/pull/9787#discussion_r455393986) [](ancestors = 455393986)